src/App.tsx

```diff
@@ -8,37 +8,119 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { client } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [isCommentButtonClicked, setIsCommentButtonClicked] = useState(false);
+
+  const displayNoPostsMessage = !posts.length
+  && selectedUser
+  && !errorMessage
+  && !isLoading;
+
+  const getAllUsers = () => {
+    return client.get<User[]>('/users');
+  };
```

Comment:
Still not fixed from the previous review, will be better if you create a separate file for all helpers functions